packages/core/src/router/routing-table.ts

```diff
@@ -140,34 +139,21 @@ export class Route {
     const pathParams = this._buildPathParams(match);
     debug(' -> found with params: %j', pathParams);
 
-    return this._createResolvedRoute(pathParams);
-  }
-
-  private _createResolvedRoute(pathParams: PathParameterValues): ResolvedRoute {
-    return this._handler ?
-      this._createResolvedHandlerRoute(pathParams) :
-      this._createResolvedControllerRoute(pathParams);
-  }
-
-  private _createResolvedHandlerRoute(
-    pathParams: PathParameterValues,
-  ): ResolvedHandlerRoute {
     return {
-      handler: this._handler!,
-      spec: this.spec,
+      route: this,
       pathParams: pathParams,
     };
   }
 
-  private _createResolvedControllerRoute(
-    pathParams: PathParameterValues,
-  ): ResolvedControllerRoute {
-    return {
-      controllerName: this._controllerName!,
-      methodName: this.spec['x-operation-name']!,
-      spec: this.spec,
-      pathParams: pathParams,
-    };
+  abstract updateBindings(requestContext: Context): void;
+
+  abstract invokeHandler(
+    requestContext: Context,
+    args: OperationArgs,
+  ): Promise<OperationRetval>;
```

Comment:
`sequence.ctx` should exist at some point, so injecting ctx might not be needed, let's discuss during one of our sync ups.